File: Classes/scoreboard.py
import json
import os

SCOREBOARD_FILE = "./Data/scoreboard.json"
MAX_SCORES = 5


def load_scores() -> list[int]:
    if not os.path.exists(SCOREBOARD_FILE):
        return []
    with open(SCOREBOARD_FILE) as f:
        return json.load(f)


def save_scores(scores: list[int]) -> None:
    with open(SCOREBOARD_FILE, "w") as f:
        json.dump(scores, f)


def add_score(score: int) -> int | None:
    scores = load_scores()
    scores.append(score)
    scores.sort(reverse=True)
    scores = scores[:MAX_SCORES]
    save_scores(scores)
    rank = scores.index(score)
    if rank < MAX_SCORES:
        return rank + 1
    return None

File: src/scoreboard.rs
use std::fs;
use std::path::Path;

const SCOREBOARD_FILE: &str = "data/scoreboard.json";
const MAX_SCORES: usize = 5;

pub fn load() -> Vec<i32> {
    if !Path::new(SCOREBOARD_FILE).exists() {
        return Vec::new();
    }
    let raw = match fs::read_to_string(SCOREBOARD_FILE) {
        Ok(s) => s,
        Err(_) => return Vec::new(),
    };
    serde_json::from_str(&raw).unwrap_or_default()
}

pub fn save(scores: &[i32]) {
    if let Ok(json) = serde_json::to_string(scores) {
        let _ = fs::write(SCOREBOARD_FILE, &json);
    }
}

pub fn add_score(score: i32) -> Option<usize> {
    let mut scores = load();
    scores.push(score);
    scores.sort_by(|a, b| b.cmp(a));
    scores.truncate(MAX_SCORES);
    save(&scores);
    scores.iter().position(|&s| s == score).map(|i| i + 1)
}
